evalml/automl/automl_search.py

```diff
@@ -652,12 +646,15 @@ def __init__(
                     self.sampler_method,
                     self.sampler_balanced_ratio,
                 )
-            if self._sampler_name not in parameters and self._sampler_name is not None:
-                parameters[self._sampler_name] = {
+            if (
```

Comment:
Hmm, I think this would be a weird move. We use a lot of information that isn't massed to the `AutoMLAlgorithm` to determine whether we use a sampler and which sampler to use. We would need to pass all of this relevant data to the `AutoMLAlgorithm` in order to move this logic, and I'm not sure if that's worth it.

Reply:
Yea If the long term plan is to move pipeline building logic to the algorithms then I think the logic for determining whether or not to add a sampler should move to the algorithms. I think there are some unused parameters in the automl algos right now that can be cleaned up too, e.g. `number_features`. We can do that in a separate issue.

Reply:
Filed issue [here](https://github.com/alteryx/evalml/issues/3400)

Reply:
thank you!